plugins/inputs/bond/bond.go

```diff
@@ -133,48 +260,59 @@ func (bond *Bond) gatherSlavePart(bondName string, rawFile string, acc telegraf.
 		name := strings.TrimSpace(stats[0])
 		value := strings.TrimSpace(stats[1])
 		if strings.Contains(name, "Slave Interface") {
-			slave = value
+			tags["interface"] = value
+			slaveCount++
 		}
-		if strings.Contains(name, "MII Status") {
-			status = 0
-			if value == "up" {
-				status = 1
-			}
+		if strings.Contains(name, "MII Status") && value == "up" {
+			fields["status"] = 1
 		}
 		if strings.Contains(name, "Link Failure Count") {
 			count, err := strconv.Atoi(value)
 			if err != nil {
 				return err
 			}
-			fields := map[string]interface{}{
-				"status":   status,
-				"failures": count,
+			fields["failures"] = count
+			if !scanPast {
+				acc.AddFields("bond_slave", fields, tags)
 			}
-			tags := map[string]string{
-				"bond":      bondName,
-				"interface": slave,
+		}
+		if strings.Contains(name, "Actor Churned Count") {
+			count, err := strconv.Atoi(value)
+			if err != nil {
+				return err
+			}
+			fields["actor_churned"] = count
+		}
+		if strings.Contains(name, "Partner Churned Count") {
+			count, err := strconv.Atoi(value)
+			if err != nil {
+				return err
 			}
+			fields["partner_churned"] = count
+			fields["total_churned"] = fields["actor_churned"].(int) + fields["partner_churned"].(int)
 			acc.AddFields("bond_slave", fields, tags)
-			slaveCount++
 		}
 	}
-	fields := map[string]interface{}{
-		"count": slaveCount,
-	}
-	tags := map[string]string{
+	tags = map[string]string{
 		"bond": bondName,
 	}
+	fields = map[string]interface{}{
+		"count": slaveCount,
+	}
 	acc.AddFields("bond_slave", fields, tags)
 
 	return scanner.Err()
 }
 
-// loadPath can be used to read path firstly from config
+// loadPaths can be used to read path firstly from config
 // if it is empty then try read from env variable
-func (bond *Bond) loadPath() {
+func (bond *Bond) loadPaths() {
 	if bond.HostProc == "" {
 		bond.HostProc = proc(envProc, defaultHostProc)
 	}
+	if bond.HostSys == "" {
+		bond.HostSys = proc(envSys, defaultHostSys)
+	}
 }
```

Comment:
This should probably live in a `func (bond *Bond) Init() error` function automatically called by telegraf after initializing the configuration fields. But it's not mandatory for this PR.

Reply:
This was just me following the existing patterns the plugin established.